cluster-autoscaler/simulator/predicatechecker/schedulerbased.go

```diff
@@ -184,6 +200,21 @@ func (p *SchedulerBasedPredicateChecker) CheckPredicates(clusterSnapshot cluster
 	return nil
 }
 
+func (p *SchedulerBasedPredicateChecker) frameworkForPod(pod *apiv1.Pod) (schedulerframework.Framework, error) {
+	fwk, ok := p.profiles[p.schedulerNameForPod(pod)]
+	if !ok {
+		return nil, fmt.Errorf("profile not found for scheduler name %q", pod.Spec.SchedulerName)
+	}
+	return fwk, nil
+}
+
+func (p *SchedulerBasedPredicateChecker) schedulerNameForPod(pod *apiv1.Pod) string {
+	if len(pod.Spec.SchedulerName) == 0 {
+		return p.defaultSchedulerName
+	}
```

Comment:
This is not useful. The apiserver always adds a default.

Reply:
ok